internal/app/decision_maker.go

```diff
@@ -124,39 +116,30 @@ func (cs *currentState) makePlan(s *state) *plan {
 				// even though I wrote it, lines like this are a code smell to me -- we need to rethink the concurrency as i mentioned above.
 				// ideally you just process all helm commands "in a background pool", they return channels,
 				// and we would be looping over select statements until we had the results of the helm commands we wanted.
-				n, e := getChartVersion(chart, version)
-				m := ""
-				if e != nil {
-					m = e.Error()
-				}
-				versionsC <- [4]string{chart, version, n, m}
+				i, e := getChartInfo(chart, version)
+				infoC <- struct {
```

Comment:
maybe instead of a channel we can just use a `sync.Mutex` and update the `extractedChartInfo` map here directly.

Reply:
done